characterEntities/Entity.java

```diff
@@ -203,6 +207,9 @@ public void setFacingEast(boolean facingEast) {
 	}
 
 	public void setAnimation(int index, AbilityAnimation abilityAnimation) {
+		if (abilityAnimations.size() > index) {
```

Comment:
this statement will probably never be true, but if it is, won't it cause an IndexOutOfBoundsException?